graphql_compiler/tests/test_compiler.py

```diff
@@ -50,6 +24,61 @@ def setUp(self):
         self.maxDiff = None
         self.schema = get_schema()
 
+        _, sqlalchemy_metadata = get_animal_schema_sql_metadata()
+        self.sql_metadata = SqlMetadata(sqlite.dialect.name, sqlalchemy_metadata)
+
+    def check_test_data(self, test_case, test_data, expected_match, expected_gremlin, expected_sql):
```

Comment:
This should either be a method on `CompilerTests`, or take `test_case` (i.e. the `self` at call sites) as the first argument, but not both.

This is also quite a large and unrelated change to this diff, which makes it harder to review this PR since the scope of the PR just broadened to include this, which means more comments and more round trips etc.

Reply:
Reverted for now.